app/models/invitiation_message_builder.rb

```diff
@@ -0,0 +1,68 @@
+class InvitationMessageBuilder
+  attr_reader :successes,
+              :errors
+
+  def initialize(no_email = false)
+    @no_email = no_email
+    @successes = []
+    @errors = []
+    @invitation
+  end
+
+  def build(message)
+    if @invitation
+      build_messages(message)
+    end
+  end
+
+  def with(invitation)
+    if invitation.is_a?(Invitation)
+      @invitation = invitation
+      self
+    end
+  end
+
+  def error_message
+    errors.join('\n')
+  end
+
+  def success_message
+    if successes? && no_email?
+      'An invitation has been created for: ' + successes.join(', ')
+    elsif successes?
+      'An invitation has been sent to: ' + successes.join(', ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.